### PR TITLE
Moved contents of bundle-workflow to the root.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@dblock.org>

### Description

This repository is dedicated to the building a complete OpenSearch distribution, thus it doesn't make sense to have a `bundle-workflow` directory. This PR moves it up, fixes all the paths, updates READMEs, etc. There are no code changes here, it's mostly a `git mv`, plus adjusted paths and merged README.

I was able to run a clean build of OpenSearch 1.1 with these changes.
 
### Issues Resolved

Closes https://github.com/opensearch-project/opensearch-build/issues/448.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ONBOARDING.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ This document describes steps to onboard a new plugin to release workflow for co
 
 3. Ensure your `build.sh` reads and passes along both `-Dbuild.snapshot=` and `-Dopensearch.version=` flags.  Snapshot builds should produce a -SNAPSHOT tagged artifact for example `opensearch-plugin-1.1.0.0-SNAPSHOT.zip` where a release build of the same component would produce `opensearch-plugin-1.1.0.0.zip`.
 
-4. Execute `./bundle-workflow/build.sh` to ensure your component builds and all artifacts are correctly placed into ./artifacts/ with correct output names.
+4. Execute `./build.sh` to ensure your component builds and all artifacts are correctly placed into ./artifacts/ with correct output names.
 
-5. Execute `./bundle-workflow/assemble.sh` to ensure the full bundle is assembled and placed in to /bundles/*.tar.gz.  Unpack the tarball to ensure all your components are placed in their correct locations.
+5. Execute `./assemble.sh` to ensure the full bundle is assembled and placed in to /bundles/*.tar.gz.  Unpack the tarball to ensure all your components are placed in their correct locations.
 
 6. Publish a PR to this repo including the updated manifest and the names of the artifacts being added.
```
Comment (peternied) [inline ONBOARDING.md:17]: Some blobs references that will be invalid with the path changes

**Review by peternied — COMMENTED** [review]
So many changes!

File: ONBOARDING.md (hunk 1 above)
Comment (dblock) [reply ONBOARDING.md:17]: I think I changed all paths. Where do you see a broken one?

Comment (dblock) [reply ONBOARDING.md:17]: Oh found'em, made relative paths to fix.

**Review by peternied — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
I will hold this until 1.1.0 release is out in case any issues caused in the process.
Otherwise I have no more objection here.
Thanks.

**Comment by peternied** [comment]
> I will hold this until 1.1.0 release is out in case any issues caused in the process. Otherwise I have no more objection here. Thanks.

Rather than block this PR, could we lock on the jenkins build job to a commit/branch? @peterzhu1992

**Comment by peterzhuamazon** [comment]
> > I will hold this until 1.1.0 release is out in case any issues caused in the process. Otherwise I have no more objection here. Thanks.
> 
> Rather than block this PR, could we lock on the jenkins build job to a commit/branch? @peterzhuamazon

We will create branch/tag after the launch.
And I dont think it is a good idea to introduce this big of a structure change right now.

**Comment by dblock** [comment]
I don't feel strongly that we should be branching, so happy to hold this till we release 1.1.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -70,8 +70,7 @@ npm install -g yarn
 Install dependencies. 
 
 ```
```
Comment (saratvemulapalli) [inline DEVELOPER_GUIDE.md:72]: There are few left overs of bundle-workflow in TOC and in some sections.

**Review by saratvemulapalli — APPROVED** [review]
The changes look good to me.
Thanks @dblock !

**Pull request merged**